adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapter.java

```diff
@@ -14,27 +14,42 @@
 package org.eclipse.hono.adapter.mqtt.impl;
 
 import java.net.HttpURLConnection;
+import java.util.HashMap;
 
 import org.eclipse.hono.adapter.mqtt.AbstractVertxBasedMqttProtocolAdapter;
 import org.eclipse.hono.adapter.mqtt.MqttContext;
 import org.eclipse.hono.adapter.mqtt.MqttProtocolAdapterProperties;
 import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.config.MapperEndpoint;
 import org.eclipse.hono.service.metric.MetricsTags;
 import org.eclipse.hono.util.Constants;
+import org.eclipse.hono.util.RegistrationConstants;
 import org.eclipse.hono.util.ResourceIdentifier;
 
 import io.netty.handler.codec.mqtt.MqttQoS;
 import io.vertx.core.Future;
 import io.vertx.core.Promise;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.mqtt.messages.MqttPublishMessage;
 
 /**
  * A Vert.x based Hono protocol adapter for publishing messages to Hono's Telemetry and Event APIs using MQTT.
  */
 public final class VertxBasedMqttProtocolAdapter extends AbstractVertxBasedMqttProtocolAdapter<MqttProtocolAdapterProperties> {
 
+    private static final String HEXES = "0123456789ABCDEF";
+
+    static String getHex(final byte[] raw) {
```

Comment:
better use *com.google.common.io.BaseEncoding.base16().encode(byte[])*

Reply:
Removed this code as we are now sending the buffer. No hex needed